.github/workflows/ci.yml

```diff
@@ -44,6 +43,54 @@ jobs:
             - name: Run Tests
               run: make test
 
+    sonatype:
+        name: Sonatype Lifecycle
+        runs-on: ubuntu-latest
+        timeout-minutes: 5
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+
+            - name: Setup Node
+              uses: actions/setup-node@v4
+              with:
+                  node-version: '18.x'
+
+            - name: Setup Go
+              uses: actions/setup-go@v5
+              with:
+                  go-version-file: 'go.mod'
+                  cache: true
+
+            # See https://help.sonatype.com/en/npm-application-analysis.html
+            - name: Install Production Node Dependencies
+              run: corepack yarn install --prod
+
+            - name: Login to Docker Hub
+              uses: docker/login-action@v3
+              with:
+                  username: ${{ secrets.DOCKER_HUB_USERNAME }}
```

Comment:
I still ran into local build issues with my own user/token for docker hub here, but guessing that's a "me" problem